pkg/api/query/v1.go

```diff
@@ -234,6 +244,22 @@ func (qapi *QueryAPI) parseEnableDedupParam(r *http.Request) (enableDeduplicatio
 	return enableDeduplication, nil
 }
 
+func (qapi *QueryAPI) parseEngineParam(r *http.Request) (queryEngine v1.QueryEngine, _ *api.ApiError) {
+	var engine v1.QueryEngine
+
+	param := promqlEngineType(r.FormValue("engine"))
+	switch param {
+	case promqlEnginePrometheus:
+		engine = qapi.queryEngine
+	case promqlEngineThanos:
+		engine = qapi.thanosEngine
+	default:
+		engine = qapi.queryEngine
```

Comment:
Ditto, let's error out with an ApiError in default case, where param is not empty, but also doesn't match valid values.